Analysis/src/TypeInfer.cpp

```diff
@@ -5030,7 +5030,8 @@ void TypeChecker::resolve(const IsAPredicate& isaP, ErrorVec& errVec, Refinement
                 return isaP.ty;
         }
 
-        return std::nullopt;
+        std::optional<TypeId> res = std::nullopt;
+        return res;
```

Comment:
...what?

Reply:
It was a wild guess and the wild guess... worked?

The error message for reference:
```
Analysis/src/TypeInfer.cpp: In static member function ‘static _Res std::_Function_handler<_Res(_ArgTypes ...), _Functor>::_M_invoke(const std::_Any_data&, _ArgTypes&& ...) [with _Res = std::optional<const Luau::TypeVar*>; _Functor = Luau::TypeChecker::resolve(const Luau::IsAPredicate&, Luau::ErrorVec&, Luau::RefinementMap&, const ScopePtr&, bool)::<lambda(Luau::TypeId)>; _ArgTypes = {const Luau::TypeVar*}]’:
Analysis/src/TypeInfer.cpp:5033:21: error: ‘<anonymous>’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
 5033 |         return std::nullopt;
      |                     ^~~~~~~
```